python/tank/platform/interactive_authentication.py

```diff
@@ -0,0 +1,195 @@
+from tank.util import authentication
+from tank.util.login import get_login_name
+from tank.util import shotgun
+
+# Configure logging
+logger = logging.getLogger("sgtk.interactive_authentication")
+# logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+
+def _get_session_token(hostname, login, password, http_proxy):
+    """
+    Retrieves a session token for the given credentials.
+    :param hostname: The host to connect to.
+    :param login: The user to get a session for.
+    :param password: Password for the user.
+    :param http_proxy: Proxy to use. Can be None.
+    :returns: If the credentials were valid, returns a session token, otherwise returns None.
+    """
+    try:
+        return authentication.generate_session_token(hostname, login, password, http_proxy)
+    except TankAuthenticationError:
+        print "Authentication failed."
+        return None
+
+
+def _do_ui_based_login(hostname, login, http_proxy):
+    pass
+
+
+def _get_keyboard_input(label, default_value=""):
+    """
+    Queries for keyboard input.
+    :param label: The name of the input we require.
+    :param default_value: The value to use if the user has entered no input.
+    :returns: The user input or default_value if nothing was entered.
+    """
+    text = label
+    if default_value:
+        text += ' [Default: %s]' % default_value
+    text += ": "
+    user_input = None
+    while not user_input:
+        user_input = raw_input(text) or default_value
+    return user_input
+
+
+def _get_user_credentials_from_keyboard(hostname, login):
+    """
+    Reads the user credentials from the keyboard.
+    :param hostname: Name of the host we will be logging on.
+    :param login: Default value for the login.
+    :returns: A tuple of (login, password) strings.
+    """
+    print "Please enter your login credentials for %s" % hostname
+    login = _get_keyboard_input("Login", login)
+    password = getpass()
+    return login, password
+
+
+def _get_user_pass_from_keyboard(hostname, login):
+    """
+    Reads the user password from the keyboard.
+    :param hostname: Name of the host we will be logging on.
+    :param login: Current user
+    :returns: The user's password.
+    """
+    print "%s, your current session has expired." % login
+    print "Please enter your password to renew your session for %s" % hostname
+    password = getpass()
+    return password
+
+
+def _do_console_based_login(hostname, default_login, http_proxy):
+    """
+    Logs to the Shotgun site using username/password combo. Retries if the
+    login attempt failed. Everything is done on the command line.
+    :param default_hostname: Default value for the hostname.
+    :param default_login: Default value for the login.
+    :param http_proxy: Proxy to use when connecting.
+    :returns: A valid Shotgun instance.
+    """
+    while True:
+        # Get the credentials from the user
+        login, password = _get_user_credentials_from_keyboard(
```

Comment:
There used to be more parameters and I broke it down to make it more readable, but it doesn't need to anymore.